fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
-                        } else {
-                            IconButton(onClick = {}) {
-                                Icon(
-                                    painter = painterResource(it.iconId),
-                                    stringResource(id = it.descriptionResourceId),
-                                    tint = FirefoxTheme.colors.iconPrimary,
-                                )
-                            }
-                        }
-                    }
-                }
-            }
-        }
+        BackButton(
+            onBackButtonClick = onBackButtonClick,
+            onBackButtonLongPress = onBackButtonLongPress,
+        )
+
+        ForwardButton(
+            onForwardButtonClick = onForwardButtonClick,
+            onForwardButtonLongPress = onForwardButtonLongPress,
+        )
+
+        HomeButton(
+            onHomeButtonClick = onHomeButtonClick,
+        )
+
+        TabsButton(
+            onTabsButtonClick = onTabsButtonClick,
+            tabCount = tabCount,
+        )
+
+        MenuButton(menuButton = menuButton)
     }
 }
 
 /**
- * Represents a navigation bar element.
+ * Top-level UI for displaying the navigation bar.
  *
- * @property iconId Resource ID of the icon that item should display.
- * @property descriptionResourceId Text used as a content description by accessibility services.
- * @property type Type of the item, defaults to [ItemType.STANDARD].
+ * @param onSearchButtonClick
+ * @param menuButton
+ * @param onTabsButtonClick
+ * @param isPrivateMode
+ * @param browserStore
  */
-data class ActionItem(
-    val iconId: Int,
-    val descriptionResourceId: Int,
-    val type: ItemType = ItemType.STANDARD,
-)
+@Composable
+fun HomeNavBar(
+    onSearchButtonClick: () -> Unit,
+    menuButton: MenuButton,
+    onTabsButtonClick: () -> Unit,
+    isPrivateMode: Boolean,
+    browserStore: BrowserStore,
+) {
+    val tabCount = browserStore.observeAsState(initialValue = 0) { browserState ->
+        if (isPrivateMode) {
+            browserState.privateTabs.size
+        } else {
+            browserState.normalTabs.size
+        }
+    }.value
 
-/**
- * Enumerates the types of items that can be used in a navigation bar.
- *
- * [STANDARD] - Represents a regular navigation item. Used for most navigation actions.
- * [TAB_COUNTER] - Represents a specialized item used to display a count, such as the number of open tabs in a browser.
- */
-enum class ItemType {
-    STANDARD, TAB_COUNTER, MENU
+    Row(
+        modifier = Modifier
+            .background(FirefoxTheme.colors.layer1)
+            .height(48.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        horizontalArrangement = Arrangement.SpaceBetween,
+    ) {
+        BackButton(
+            onBackButtonClick = {},
+            onBackButtonLongPress = {},
+        )
+
+        ForwardButton(
+            onForwardButtonClick = {},
+            onForwardButtonLongPress = {},
```

Comment:
for these yet-to-be implemented lambdas, let's go ahead and add them to the nav bar's API and then have them as empty lambdas at their callsite in the fragment. Our goal with this PR will be to get a stable, 99%-done API for the Nav Bar UI, and then future PRs will simply plug-in the business logic in the fragments.

Reply:
actually, in this case there will be no click handling. in HomeNavBar buttons will be always disabled so no need for click handlers